pkg/component/base/execution.go

```diff
@@ -29,28 +29,36 @@ type IExecution interface {
 	Execute(context.Context, []*Job) error
 }
 
+// Job is the job for the component.
 type Job struct {
 	Input  InputReader
 	Output OutputWriter
 	Error  ErrorHandler
 }
 
+// InputReader is an interface for reading input data from a job.
```

Comment:
Thanks for this, my IDE (also) complains about uncommented exported types and functions. I've tried to remove such warnings as I modify packages as you're doing with this PR 👍 👍